src/platform/Infineon/PSOC6/ConnectivityManagerImpl.cpp

```diff
@@ -446,10 +446,18 @@ void ConnectivityManagerImpl::DriveAPState()
                 // Compute the amount of idle time before the AP should be deactivated and
                 // arm a timer to fire at that time.
                 System::Clock::Timeout apTimeout = (mLastAPDemandTime + mWiFiAPIdleTimeout) - now;
-                err                              = DeviceLayer::SystemLayer().StartTimer(apTimeout, DriveAPState, nullptr);
-                SuccessOrExit(err);
                 ChipLogProgress(DeviceLayer, "Next WiFi AP timeout in %" PRIu32 " ms",
                                 System::Clock::Milliseconds32(apTimeout).count());
+                SystemLayer().ScheduleLambda([apTimeout, this] {
+                    CHIP_ERROR ret = CHIP_NO_ERROR;
+                    ret            = DeviceLayer::SystemLayer().StartTimer(apTimeout, DriveAPState, this);
+                    SuccessOrExit(ret);
```

Comment:
This `SuccessOrExit` is redundant here if you are checking `if (ret != CHIP_NO_ERROR)` 2 lines below. For readability maybe it's better to use:
```c++
VerifyOrReturn(ret != CHIP_NO_ERROR, ChipLogError(NotSpecified, "StartTimer failed %s: ", chip::ErrorStr(ret)));
```

The same thing with lambda below.

Reply:
@arkq  : thanks for the review. I have updated the code.